javascript/README.md

```diff
@@ -0,0 +1,293 @@
+[prefer-object-spread]: https://eslint.org/docs/rules/prefer-object-spread
+[quote-props]: https://eslint.org/docs/rules/quote-props.html
+
+# JavaScript Guide
+
+*A mostly reasonable approach to JavaScript, inspired by the [Airbnb JavaScript Style Guide].*
+
+## Table of Contents
+
+1. [References](#references) (Variables, Constants, etc.)
+2. [Objects](#objects)
+
+---
+
+## References
+
+### 1.1 Prefer Constants
+
+Use `const` for all of your references; avoid using `var`. 
+
+> Why? This ensures that you can’t reassign your references, which can lead to bugs and difficult to comprehend code.
+
+ESLint: [prefer-const], [no-const-assign]
+
+🚫 Nope. 🚫
+
+```js
+var a = 1;
+var b = 2;
+```
+
+🎉 Yep! 🎉
+
+```js
+const a = 1;
+const b = 2;
+```
+
+### 1.2 Reassigning References
+
+If you must reassign references, use `let` instead of `var`. 
+
+> Why? `let` is block-scoped rather than function-scoped like `var`. Function-scoped variables are hoisted which can lead to bugs if you are not careful. Using block-scoped variables makes our code more predictable by giving the variable an explicit scope.
+
+ESLint: [no-var]
+
+🚫 Nope. 🚫
+
+```js
+var count = 1;
+if (true) {
+  count += 1;
+}
+```
+
+🎉 Yep! 🎉
+
+```js
+let count = 1;
+if (true) {
+  count += 1;
+}
+```
+
+### 1.3 Block Scope
+
+Note that both `let` and `const` are block-scoped.
+
+```js
+// Both `const` and `let` only exist in the blocks they are defined in.
+{
+  let a = 1;
+  const b = 1;
+}
+console.log(a); // ReferenceError: a is not defined
+console.log(b); // ReferenceError: b is not defined
+```
+
+[⇧ top](#javascript-guide)
+
+---
+
+## Objects
+
+### 2.1 Object Creation 
+
+Use the literal syntax for object creation. 
+
+> While there are no performance differences between the two approaches, the byte savings and conciseness of the object literal form is what has made it the de facto way of creating new objects.
+
+ESLint: [no-new-object]
+
+🚫 Nope. 🚫
+
+```js
+const item = new Object();
+````
+
+🎉 Yep! 🎉
+
+```js
+const item = {};
+```
+
+### 2.2 Object Shorthand
+  
+Use object shorthand for both methods and property values. 
+
+> ECMAScript 6 provides a concise form for defining object literal methods and properties. This syntax can make defining complex object literals much cleaner.
+
+ESLint: [object-shorthand]
+
+#### Object Method
+
+🚫 Nope. 🚫
+
+```js
+const atom = {
+  value: 1,
+  addValue: function (value) {
+    return atom.value + value;
+  },
+};
+```
+
+🎉 Yep! 🎉
+
+```js
+const atom = {
+  value: 1,
+  addValue(value) {
+    return atom.value + value;
+  },
+};
+```
+
+#### Object Property
+
+🚫 Nope. 🚫
+
+```js
+const generalLeiaOrgana = 'General Leia Organa';
+
+const obj = {
+  generalLeiaOrgana: generalLeiaOrgana,
+};
+```
+
+🎉 Yep! 🎉
+
+```js
+const generalLeiaOrgana = 'General Leia Organa';
+
+const obj = {
+  generalLeiaOrgana,
+};
+```
+
+### 2.3 Group Object Shorthand Properties
+
+Group your shorthand properties at the beginning of your object declaration.
```

Comment:
This is new to me. I feel like we won't be able to consistently follow this rule without an ESLint rule going along with it. Also (maybe this is just me) but I think that grouping object properties based on there being a shorthand seems kind of arbitrary.

Reply:
I kind of felt that way when seeing it. I think there could be a more defined rule that says don't mix the two (shorthand and non-shorthand), but don't really think that's practical, either.

Reply:
@calebeby It is arbitrary. 😝 

I don't imagine all JS guide suggestions will have an ESLint rule to apply. If folks feel strongly against this suggestion, make your case. 😉

Reply:
```js
const asdf = (middleName, birthDate, id) => {
  return {
    firstName: names[id].first,
    middleName,
    lastName: names[id].last,
    birthDate,
  }
}
```

Silly example, but I think this shows how I think it makes more sense to group object keys by similar concepts rather than by whether they are shorthands. `middleName` conceptually fits more closely with `firstName` and `lastName` than with `birthDate`

Reply:
@calebeby Thanks for the example! Great feedback! I do agree with your example. 👍 

Do you think I should change the `2.3` section to say "Order Object Object Keys By Concept"? Or would you suggest I remove that section all together?

Reply:
I think you can remove it

Reply:
> I kind of felt that way when seeing it. I think there could be a more defined rule that says don't mix the two (shorthand and non-shorthand), but don't really think that's practical, either.

@Josh68 I'm going to push back on this one. I've had plenty times where both come into play and I don't think it makes sense to _not_ use object shorthand keys if you can. :)

Reply:
> I think you can remove it

I'm okay with this. Thanks for the discussion! 😉